src/v/cloud_storage/tests/manual_fixture.h

```diff
@@ -17,28 +17,51 @@
 #include "redpanda/tests/fixture.h"
 #include "storage/disk_log_impl.h"
 
+// Mixin to disable cloud storage background work, for use before a fixture is
+// initialized, in tests that want manual control over remote operations.
+class manual_cloud_storage_mixin {
+public:
+    manual_cloud_storage_mixin() {
+        config::shard_local_cfg()
+          .cloud_storage_enable_segment_merging.set_value(false);
+        config::shard_local_cfg()
+          .cloud_storage_disable_upload_loop_for_tests.set_value(true);
+    }
+    ~manual_cloud_storage_mixin() {
+        config::shard_local_cfg()
+          .cloud_storage_enable_segment_merging.set_value(true);
+        config::shard_local_cfg()
+          .cloud_storage_disable_upload_loop_for_tests.set_value(false);
```

Comment:
nit: I think `property.reset()` is preferable for this type of situations since it doesn't assume what the default values is.

Reply:
ditto for `manual_storage_mixin`

Reply:
Done